pythran/tests/test_typing.py

```diff
@@ -66,6 +66,44 @@ def typing_aliasing_and_update():
     return a'''
         self.run_test(code, typing_aliasing_and_update=[])
 
+    def test_typing_aliasing_and_update_and_globals(self):
+        code = '''
+def f(l): return len(l)
+def g(l): l.append(1) ; return 0
+foo=[f,g]
+def typing_aliasing_and_update_and_globals(i):
+    h = []
+    return foo[i](h), h'''
```

Comment:
> There are no issue with order of evaluation for comma separated instruction?

gotcha! very smart one :-)

It turns out it's ok here because we're
handling references, so whatever the evaluation order, in the end we get an int and a reference to a list.

If we evaluate from left to right, we first update the list then return
a reference to this list.

if we evaluate from right to left we first return a reference then
evaluate then update this reference.

From a call site point of view, we get an int and a reference, and the
reference ultimately points to the updated list ;-)